docs/en/week08/08-3.md

```diff
@@ -0,0 +1,198 @@
+$$\theta_{Dec} = \theta_{Dec} - \triangledown_{\theta_{Dec}}C(y,\tilde{y})\\ \\
+\theta_{Enc} = \theta_{Enc} - \triangledown_{\theta_{Enc}}D(\check{z},\tilde{z})
+$$
+In this way $z$ is constrained in taking only a set of values.
+
+### Autoencoder (AE)
+<center>
+<img src="{{site.baseurl}}/images/week08/08-3/AE.png" /><br>
+<b>Fig. 1</b>: Autoencoder
+</center>
+
+As shown in *Fig. 1*, to summarize at a high level, a very simple form of AE is as follows:
+
+- First, the autoencoder takes in an input and maps it to a hidden state through an affine transformation $\boldsymbol{h} = f(\boldsymbol{W}_h \boldsymbol{x} + \boldsymbol{b}_h)$, where $f$ is an (element-wise) activation function. This is the **encoder** stage. Note that $\boldsymbol{h}$ is also called the **code**.
+- Next, $\hat{\boldsymbol{x}} = g(\boldsymbol{W}_x \boldsymbol{h} + \boldsymbol{b}_x)$, where $g$ is an activation function. This is the **decoder** stage.
+
+**Reconstruction costs**
+
+When the input is real-valued we use mean squared error loss, 
+$$
+C(y,\tilde{y}) = \mid \mid{y} - \tilde{y} \mid \mid^{2} =  \mid\mid{y - \text{Dec}[\text{Enc}(\tilde{y})]}\mid\mid^{2}
+$$
+
+When the input is categorical we use cross entropy loss,
+$$
+C(y,\tilde{y}) = -\sum_{i = 1}^{n} [y_{i} \log(\tilde{y_{i}}) + (1 - y_{i})\log(1 - \tilde{y_{i}})]
+$$
+
+**Loss functional**
+
+The loss functional is the average of the per sample loss.
+
+$$
+\mathcal{L}(F(\cdot), Y) = \frac{1}{m} \sum_{j = 1}^{m} \mathcal{l}(F(\cdot),y^{(j)}) \in \mathbb{R} \\
+\mathcal{l}_\text{energy}(F(\cdot),y) = F(y)
+$$
+
+**Under-/over-complete hidden layer**
+
+When the dimensionality of the hidden layer is less than the dimensionality of the input then it is under complete hidden layer. When the dimensionality of the hidden layer is greater than the dimensionality of the input then it is over complete. 
+
+The dimensionality of the hidden layer is larger than the input so that they can be linearly seperable. However, this can lead to a collapsed representation. Since by reconstructing the input, the model can copy all the features. In order to avoid this there are a few techniques such as denoising autoencoder, contractive autoencoder, variation autoencoder, etc. 
+
+### Denoising Autoencoder
+<center>
+<img src="{{site.baseurl}}/images/week08/08-3/DAE.png" /><br>
+<b>Fig. 2</b>: Denoising Autoencoder
+</center>
+
+Denoising Autoencoder is a contrastive technique. As shown in *Fig. 2*, we corrupt the input data $y$, and get the sample $\hat y$ which has a high energy. The energy associated to $\tilde y$ is squared euclidian distance from its original value. Then $\tilde y$ is encoded back to the hidden variable $h$ and then passed back to the decoder producing $\tilde y$ which should be close to the target $y$. Hence irrespective of what noise is added to the original data, the autoencoder needs to learn to separate the noise and retrieve the original value of $y$.
```

Comment:
Please, use `#fb8072` for the red hot `vect{\hat{y}}`  and `#80b1d3` for the blue cold `\vect{y}`.

Reply:
done